modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/ObjectEntryModelListener.java

```diff
@@ -396,6 +421,98 @@ private Map<String, Object> _toDTO(ObjectEntry objectEntry, User user)
 		return objectEntry.getModelAttributes();
 	}
 
+	private void _updateObjectViewFilterColumn(
+		String newExternalReferenceCode, ObjectEntry objectEntry) {
+
+		String sql = StringBundler.concat(
+			"select ObjectViewFilterColumn.objectViewFilterColumnId, ",
+			"ObjectViewFilterColumn.objectViewId from ObjectViewFilterColumn ",
+			"inner join ObjectField on ObjectViewFilterColumn.objectFieldName ",
+			"= ObjectField.name inner join ObjectRelationship on ",
+			"ObjectField.objectFieldId = ObjectRelationship.objectFieldId2 ",
+			"inner join ObjectDefinition on ",
+			"ObjectRelationship.objectDefinitionId1 = ",
+			"ObjectDefinition.objectDefinitionId where ObjectDefinition.",
+			"objectDefinitionId = ?");
+
+		Connection connection = _currentConnection.getConnection(
+			_objectEntryPersistence.getDataSource());
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				sql)) {
+
+			preparedStatement.setLong(
+				1, GetterUtil.getLong(objectEntry.getObjectDefinitionId()));
+
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			ObjectView objectView = null;
+
+			while (resultSet.next()) {
+				objectView = _objectViewLocalService.getObjectView(
+					resultSet.getLong("objectViewId"));
+
+				List<ObjectViewFilterColumn> objectViewFilterColumns =
+					new ArrayList<>();
+
+				for (ObjectViewFilterColumn objectViewFilterColumn :
+						objectView.getObjectViewFilterColumns()) {
+
+					if (objectViewFilterColumn.getObjectViewFilterColumnId() ==
+							resultSet.getLong("objectViewFilterColumnId")) {
+
+						JSONObject jsonObject =
+							JSONFactoryUtil.createJSONObject(
+								objectViewFilterColumn.getJSON());
+
+						JSONArray jsonArray = jsonObject.getJSONArray(
+							objectViewFilterColumn.getFilterType());
+
+						JSONArray newJSONArray = new JSONArrayImpl();
+
+						for (int i = 0; i < jsonArray.length(); i++) {
+							if (StringUtil.equals(
+									(String)jsonArray.get(i),
+									objectEntry.getExternalReferenceCode())) {
+
+								if (!StringUtil.equals(
+										newExternalReferenceCode,
+										StringPool.BLANK)) {
+
+									newJSONArray.put(newExternalReferenceCode);
+								}
+							}
+							else {
+								newJSONArray.put((String)jsonArray.get(i));
+							}
+						}
+
+						if (newJSONArray.length() == 0) {
+							continue;
+						}
+
+						String json = StringBundler.concat(
+							"{\"", objectViewFilterColumn.getFilterType(),
+							"\":", newJSONArray, "}");
+
+						objectViewFilterColumn.setJSON(json);
+					}
+
+					objectViewFilterColumns.add(objectViewFilterColumn);
+				}
+
+				_objectViewLocalService.updateObjectView(
+					objectView.getObjectViewId(),
+					objectView.isDefaultObjectView(), objectView.getNameMap(),
+					objectView.getObjectViewColumns(), objectViewFilterColumns,
+					objectView.getObjectViewSortColumns());
```

Comment:
I believe we can create a method in `ObjectViewFilterColumnLocalServiceImpl` to update only the object view filter columns.

what do you think?